Services/IotHub/IotHubConnectionStringManager.cs

```diff
@@ -134,16 +146,35 @@ public void ValidateConnectionString(string connectionString)
             // if a key is provided, check if IoTHub is valid
             if (!match.Groups[CONNSTRING_REGEX_KEY].Value.IsNullOrWhiteSpace())
             {
+                RegistryManager registryManager = null;
                 try
                 {
-                    RegistryManager.CreateFromConnectionString(connectionString);
+                    registryManager = RegistryManager.CreateFromConnectionString(connectionString);
                 }
-                catch (IOException)
+                catch (Exception e)
                 {
                     string message = "Could not connect to IotHub with the connection" +
                                      "string provided. Check that the key is valid and " +
                                      "that the hub exists.";
-                    throw new IotHubConnectionException(message);
+                    throw new IotHubConnectionException(message, e);
+                }
+
+                try
+                {
+                    // check read permissions
+                    await registryManager.GetDevicesAsync(1, CancellationToken.None);
+
+                    // check write permissions
+                    var testDevice = new Device();
```

Comment:
what's the ID of the test device? I would give it an ID to explain what it is, in case the user sees it in the portal (note that RemoveDeviceAsync could fail and leave the device in hub).  Also, you might get a conflict if the device exists already, so I would add a suffix like "current time in milliseconds". Also consider that if RemoveDeviceAsync keeps failing, the hub will be polluted with test devices.